launch/bash.go

```diff
@@ -31,6 +32,9 @@ func (b *BashShell) Launch(proc ShellProcess) error {
 		bashCommand = bashCommandWithTokens(len(proc.Args) + 1)
 	}
 	launcher += bashCommand
+	if err := os.Chdir(proc.WorkingDirectory); err != nil {
```

Comment:
This would also change the working directory for profile.d scripts. I think instead we'll want to throw a `cd workdir` right about here: https://github.com/buildpacks/lifecycle/blob/6f56a153034317a88f41109b76e026b93134eb89/launch/bash.go#L33